File: lib/app_count_web/channels/users/user_socket.ex
defmodule AppCountWeb.Users.UserSocket do
  use Phoenix.Socket

  channel("rewards:*", AppCountWeb.Users.RewardsChannel)

  def connect(params, socket) do
    case AppCount.Accounts.Utils.Tokens.verify_token(params["user_token"]) do
      {:error, :bad_token} -> :error
      {:ok, user, _} -> {:ok, assign(socket, :user, %{id: user.id})}
    end
  end

  def id(socket) do
    "user_socket:#{socket.assigns.user.id}"
  end
end

File: assets/js/utils/userSocket.js
import {Socket} from 'phoenix';

const socket = new Socket('/ws/users', {});
socket.connect();
let channel = socket.channel(`users/${window.current_user_id}`, {})
channel.join()
    .receive("ok", resp => { console.log("Joined successfully", resp) })
    .receive("error", resp => { console.log("Unable to join", resp) })

export default socket;